### PR TITLE
Feature/instance normalization

### PR DESCRIPTION
**Pull request by sbodenstein**
Implementation of the Instance Normalization layer introduced by [Instance Normalization:
The Missing Ingredient for Fast Stylization](https://arxiv.org/abs/1607.08022), and implemented by the authors [here](https://github.com/DmitryUlyanov/texture_nets/blob/master/InstanceNormalization.lua).

One question: I used the terminology "weight" and "bias" to refer to the learnable parameters. But perhaps they should be the same as used in batch norm (beta and gamma).

### CODE REVIEW

**Comment by piiswrong** [comment]
I think beta/gamma is better. Make sure it has the right correspondence

**Comment by sbodenstein** [comment]
@piiswrong: Ok, done. Should be correct now.

**Comment by winstywang** [comment]
What is the difference between this op and BN with batchsize = 1?

**Comment by sbodenstein** [comment]
@winstywang: I mention the two differences with Batch Normalization in the operators `.describe` field. But with batch size 1, there is a single difference: there is no difference between train-time and test time (so no running means and variances kept in aux states).

**Comment by winstywang** [comment]
@sbodenstein I see. It seems that it is a special case of our paper: https://arxiv.org/abs/1603.04779 :-)

**Comment by winstywang** [comment]
Let's wait travis, and then merge
